telegram/files/file.py

```diff
@@ -119,30 +123,45 @@ def download(
         if custom_path is not None and out is not None:
             raise ValueError('custom_path and out are mutually exclusive')
 
-        # Convert any UTF-8 char into a url encoded ASCII string.
-        url = self._get_encoded_url()
+        local_file = is_local_file(self.file_path)
+
+        if local_file:
+            url = self.file_path
+        else:
+            # Convert any UTF-8 char into a url encoded ASCII string.
+            url = self._get_encoded_url()
 
         if out:
-            buf = self.bot.request.retrieve(url)
-            if self._credentials:
-                buf = decrypt(
-                    b64decode(self._credentials.secret), b64decode(self._credentials.hash), buf
-                )
+            if local_file:
+                with open(url, 'rb') as file:
+                    buf = file.read()
```

Comment:
I think there are more efficient ways to copy a file than reading a file into python memory.
Maybe the shutil package can assist here.

Reply:
You're right. I think `shutil.copyfile()` would be the one to use.